rfcs/0000-error-handling.md

```diff
@@ -0,0 +1,144 @@
+}
+```
+
+So that applications will always be aware that queries may have inconsistent data. Developers should refer to the indexing status API for details on the errors.
+
+### Indexing Status API
+
+The indexing status API will be revamped.
+
+The subgraph indexing status API currently has the top level query:
+
+    indexingStatusesForSubgraphName(subgraphName: String!): [SubgraphIndexingStatus!]!
+
+This is changed to:
+
+    indexingStatusesForSubgraphName(subgraphName: String!): SubgraphIndexingStatus
+
+Which returns null if the name is not found, and returns the status for the current version of the subgraph otherwise.
+
+A field `failedAfter`  is added to record the block previous to the first error.
+
+Errors are made more than just a string with the `Error` type.
+
+The complete indexing status API schema is:
+
+```graphql
+type Query {
+  indexingStatusesForSubgraphName(subgraphName: String!): SubgraphIndexingStatus
+  indexingStatuses(subgraphs: [String!]): [SubgraphIndexingStatus!]!
+}
+
+type SubgraphIndexingStatus {
+  subgraph: String!
+  synced: Boolean!
+  errors: [Error!]! # Sorted from first to last
+  chains: [ChainIndexingStatus!]!
+  node: String!
+}
+
+# Fields are moved from `EthereumIndexingStatus` to the interface.
+interface ChainIndexingStatus {
+  network: String!
+  chainHeadBlock: Block
+  earliestBlock: Block
+  latestBlock: Block
+  failedAfter: Block
+}
+
+type EthereumIndexingStatus implements ChainIndexingStatus { }
+
+# Renamed from `EthereumBlock` to `Block`
+type Block {
+  hash: Bytes!
+  number: BigInt!
+}
+
+type Error {
+	message: String!
+	
+	# Context for the error.
+	network: String!
+	block: Block!
+	transaction: Bytes
+	handler: String
+}
+```
+
+Since the indexing API is a view to the subgraph metadata, an `Error` type is also added to the metadata where subgraph errors are recorded.
+
+## Compatibility
+
+This does incompatible changes to the indexing status API, however that API was so far experimental and undocumented.
+
+The behavior of synced, failed subgraphs is changed to keep indexing. Our consensus is that the current behavior is useful to nobody and this can only be an improvement.
+
+## Drawbacks and Risks
+
+- Changing the behavior of failed subgraphs has risks. There may be unanticipated consequences which may slip through testing. We may have gotten our rationale wrong and users may find the new behavior worse for factors we did not consider.
+
+## Alternatives
+
+- Don't change the default behavior, and put make the "continue on failure" mode opt-in through a flag on the manifest. With the possible intention of making it the default in the future. This exposes us to less risk when rolling this out. The downsides are that adoption and feedback will be slower and developers will need to update their subgraphs in order to benefit.
+
+## Open Questions
+
+- The field name `failedAfter` is up for bikeshedding, aternatives: `lastHealthyBlock`, `blockBeforeFailure`, `lastGoodBlock`...
```

Comment:
Heh, yeah, I was going to suggest `lastHealthyBlock`. 😄

Reply:
Switched to that.